packages/core/src/lib/filter/v2/index.ts

```diff
@@ -0,0 +1,338 @@
+      this.peer.id.toString(),
+      [
+        ...(FilterV2.subscriptionsLog.get(this.peer.id.toString()) ?? []),
+        subscription,
+      ] ?? [subscription]
+    );
+  }
+
+  async unsubscribe(contentTopics: ContentTopic[]): Promise<void> {
+    const stream = await this.newStream(this.peer);
+    const unsubscribeRequest = FilterSubscribeRpc.createUnsubscribeRequest(
+      this.pubSubTopic,
+      contentTopics
+    );
+
+    try {
+      await pipe([unsubscribeRequest.encode()], lp.encode, stream.sink);
+    } catch (error) {
+      throw new Error("Error subscribing: " + error);
+    }
+  }
+
+  async ping(): Promise<void> {
+    const stream = await this.newStream(this.peer);
+
+    const request = FilterSubscribeRpc.createSubscriberPingRequest();
+
+    try {
+      const res = await pipe(
+        [request.encode()],
+        lp.encode,
+        stream,
+        lp.decode,
+        async (source) => await all(source)
+      );
+
+      const { statusCode, requestId, statusDesc } =
+        FilterSubscribeResponse.decode(res[0].slice());
+
+      if (statusCode < 200 || statusCode >= 300) {
+        throw new Error(
+          `Filter ping request ${requestId} failed with status code ${statusCode}: ${statusDesc}`
+        );
+      }
+
+      log("Ping successful");
+    } catch (error) {
+      log("Error pinging: ", error);
+      throw new Error("Error pinging: " + error);
+    }
+  }
+
+  async unsubscribeAll(): Promise<void> {
+    const stream = await this.newStream(this.peer);
+
+    const request = FilterSubscribeRpc.createUnsubscribeAllRequest(
+      this.pubSubTopic
+    );
+
+    try {
+      const res = await pipe(
+        [request.encode()],
+        lp.encode,
+        stream,
+        lp.decode,
+        async (source) => await all(source)
+      );
+
+      const { statusCode, requestId, statusDesc } =
+        FilterSubscribeResponse.decode(res[0].slice());
+
+      if (statusCode < 200 || statusCode >= 300) {
+        throw new Error(
+          `Filter unsubscribe all request ${requestId} failed with status code ${statusCode}: ${statusDesc}`
+        );
+      }
+
+      log("Unsubscribed from all content topics");
+    } catch (error) {
+      throw new Error("Error unsubscribing from all content topics: " + error);
+    }
+  }
+}
+
+class FilterV2 extends BaseProtocol implements IReceiver {
+  private readonly options: ProtocolCreateOptions;
+  static subscriptionsLog: SubscriptionsLog = new Map();
+
+  constructor(public libp2p: Libp2p, options?: ProtocolCreateOptions) {
+    super(
+      FilterV2Codecs.SUBSCRIBE,
+      libp2p.peerStore,
+      libp2p.getConnections.bind(libp2p)
+    );
+
+    this.libp2p.handle(FilterV2Codecs.PUSH, this.onRequest.bind(this));
+
+    this.options = options ?? {};
+  }
+
+  async createSubscription(
+    _pubSubTopic?: string,
+    peerId?: PeerId
+  ): Promise<Subscription> {
+    const pubSubTopic =
+      _pubSubTopic ?? this.options.pubSubTopic ?? DefaultPubSubTopic;
```

Comment:
```suggestion
    pubSubTopic?: string,
    peerId?: PeerId
  ): Promise<Subscription> {
    const _pubSubTopic =
      pubSubTopic ?? this.options.pubSubTopic ?? DefaultPubSubTopic;
```

When having a dupe variable, you'd usually prefix the variable with the shortest scope, instead of exposing the prefix outside the function scope (via the function signature in this instance).